simulation/groups.py

```diff
@@ -0,0 +1,342 @@
+__all__ = [
+    'SGroup',
+    'IGroup',
+    'RGroup',
+    'DGroup'
+]
+
+from simulation import groups_interface
+import random
+
+
+class SGroup(groups_interface.PopulationGroup):
+    def __init__(self, x: float, y: float, label: str, inf_dist: float, infection_prob: float, recover_prob: float,
+                 death_prob: float) -> None:
+
+        super().__init__(x, y, label, inf_dist, infection_prob, recover_prob, death_prob)
+
+    def __del__(self):
+        pass
+
+    def move(self, x_distance: float, y_distance: float, box_width: int,
+             box_height: int) -> bool:
+
+        can_move = True
+
+        if not 0 < self.x + x_distance < box_width:
+            can_move = False
+
+        if not 0 < self.y + y_distance < box_height:
+            can_move = False
+
+        if can_move:
+            self.x += x_distance
+            self.y += y_distance
+
+            return True
+        else:
+            return False
+
+    def is_in_area(self, member_x: float, member_y: float, member_inf_dis: float) -> bool:
+        if self.x > member_x:
+            x_difference = self.x - member_x
+        else:
+            x_difference = member_x - self.x
+
+        if self.y > member_y:
+            y_difference = self.y - member_y
+        else:
+            y_difference = member_y - self.y
+
+        if x_difference < member_inf_dis or \
```

Comment:
I assume that infection distance is a radius right? So to calculate a true Cartesian distance between two points on a plane, you should rather use this formula: https://www.thoughtco.com/understanding-the-distance-formula-2312242